go/kvstore/cache.go

```diff
@@ -106,49 +106,44 @@ func (k *KVRevisionCache) Put(mctx libkb.MetaContext, entryID keybase1.KVEntryID
 	return nil
 }
 
-func (k *KVRevisionCache) checkDeletableLocked(mctx libkb.MetaContext, entryID keybase1.KVEntryID, revision int) (err error) {
+func (k *KVRevisionCache) checkForUpdateLocked(mctx libkb.MetaContext, entryID keybase1.KVEntryID, revision int) (err error) {
 	k.ensureIntermediateLocked(entryID)
 
 	entry, ok := k.data[entryID.TeamID][entryID.Namespace][entryID.EntryKey]
 	if !ok {
-		// this should never happen
-		return KVRevisionCacheError{fmt.Sprintf("cache is corrupted - please restart")}
-	}
-	entryHash := DELETED_OR_NONEXISTENT
-	if revision < entry.Revision {
-		return KVRevisionCacheError{fmt.Sprintf("cache error: revision decreased from %d to %d", entry.Revision, revision)}
+		// this entry didn't exist in the cache, so there's nothing to check
+		return nil
 	}
-	if revision == entry.Revision {
-		if entryHash != entry.EntryHash {
-			return KVRevisionCacheError{fmt.Sprintf("cache error: at the same revision (%d) hash of entry cannot be different: %s -> %s", revision, entry.EntryHash, entryHash)}
-		}
+	if revision <= entry.Revision {
+		return KVRevisionCacheError{fmt.Sprintf("expected revision greater than %d", entry.Revision)}
```

Comment:
When a user's call gets blocked because their revision is too low, that should result in an identifiable error. Looks like SCTeamStorageWrongRevision is being used for that in some tests. Doesn't this here error also have the potential to be returned for that same condition? But KVRevisionCacheError is generic.

Reply:
yeah - it seemed like the options here were:
1) not check for this condition with the cache and let those requests through to the server that we know should fail,
2) throw the same error here as the server, which very much looks like an http request error
3) throw a different error but with a really similar error message

i opted for (3) for a few reasons: (1) and (2) seemed bad. and also, this error shouldn't happen very often in a normal flow. it means the user is intentionally using a revision that they should absolutely know is too low. it just seemed like maybe it wasn't worth doing anything too weird for it. but i'm definitely open to discussion.

Reply:
OK. i changed it to (2). it turns out we're doing not-dissimilar things in other places, and doing this allowed us to leverage some of the CLI machinery to get the error codes to populate correctly. this is 👍.